src/Middleware/RateLimiting/src/LeaseContext.cs

```diff
@@ -7,12 +7,19 @@ namespace Microsoft.AspNetCore.RateLimiting;
 
 internal struct LeaseContext : IDisposable
 {
-    public bool? GlobalRejected { get; init; }
+    public RequestRejectionReason? RequestRejectionReason { get; init; }
 
-    public required RateLimitLease Lease { get; init; }
+    public RateLimitLease Lease { get; init; }
```

Comment:
Odd that the compiler isn't complaining about this, mark it nullable.

Reply:
Weird, done. If it's a point of curiosity, I built/tested on my mac, so maybe that's related